wasmplugin/plugin.go

```diff
@@ -102,11 +102,18 @@ type httpContext struct {
 	requestBodySize       int
 	responseBodySize      int
 	metrics               *wafMetrics
+	interruptionHandled   bool
 }
 
 // Override types.DefaultHttpContext.
 func (ctx *httpContext) OnHttpRequestHeaders(numHeaders int, endOfStream bool) types.Action {
 	defer logTime("OnHttpRequestHeaders", currentTime())
+
+	if ctx.interruptionHandled {
```

Comment:
Shall we handle this check in OnHttpRequestHeaders? Can something trigger an interruption before this callback?

Reply:
I really don’t think so, but I don’t know if there could be corner cases or malformed requests that may trigger this callback twice. That's the reason why I went for a conservative approach.
Still, the whole callback implementation is designed to happen just once, and in the worst case that is called again with an interruption we still have the panic inside `handleInterruption`. For these reasons, I would remove it